crypto/fipsmodule/sha/sha512.c

```diff
@@ -288,13 +333,23 @@ static int sha512_final_impl(uint8_t *out, size_t md_len, SHA512_CTX *sha) {
     return 0;
   }
 
-  assert(md_len % 8 == 0);
   const size_t out_words = md_len / 8;
+  const size_t remainder = md_len % 8;
```

Comment:
I'd prefer to be more specific here. Since we support only 512, 256, and 224 digest lengths, how about
```
assert(md_len == SHA512_LEN || md_len == SHA512_256_LEN || md_len == SHA512_224_LEN)
```
and then further down
```
if (md_len == SHA512_224_LEN) {
...
}
```

Reply:
Tightening this up makes sense. I'll update.

Reply:
on second pass, it looks like we also [use this function impl for SHA384](https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/sha/sha512.c#L188). rather than case out all of the functions that call this, I'm going to keep the assertion as it was before this PR, but extend it to also accept the `md_len == SHA512_224_LEN` case.